File: frontend/src/types/kontakt.ts
export interface Kontakt {
  id: number;
  vorname: string;
  nachname: string;
  email: string;
  telefon: string;
  debitorId: number;
  debitorName?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface KontaktFormData {
  vorname: string;
  nachname: string;
  email: string;
  telefon: string;
  debitorId: number;
}

export const emptyKontaktForm = (debitorId: number): KontaktFormData => ({
  vorname: "",
  nachname: "",
  email: "",
  telefon: "",
  debitorId,
});

export function kontaktDisplayName(kontakt: Kontakt): string {
  return [kontakt.vorname, kontakt.nachname].filter(Boolean).join(" ");
}

File: backend/internal/models/kontakt.go
package models

import "time"

type Kontakt struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Vorname   string    `gorm:"size:100" json:"vorname"`
	Nachname  string    `gorm:"size:100;not null" json:"nachname"`
	Email     string    `gorm:"size:255" json:"email"`
	Telefon   string    `gorm:"size:50" json:"telefon"`
	DebitorID uint      `gorm:"not null;index" json:"debitorId"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}
